ReactiveCocoa/Swift/SignalProducer.swift

```diff
@@ -389,6 +389,17 @@ extension SignalProducerType {
 		}
 	}
 
+	/// Lifts a binary Signal operator to operate upon a Signal and a SignalProducer instead.
+	///
+	/// In other words, this will create a new SignalProducer which will apply
+	/// the given Signal operator to _every_ Signal created from the two
+	/// producers, just as if the operator had been applied to each Signal
+	/// yielded from start().
+	@warn_unused_result(message="Did you forget to call `start` on the producer?")
+	public func lift<U, F, V, G>(transform: Signal<Value, Error> -> Signal<U, F> -> Signal<V, G>) -> Signal<U, F> -> SignalProducer<V, G> {
+		return self.lift(transform)
```

Comment:
May I ask if there is one in `SignalProducerType`? I can see only one `lift` implementation which takes a `Signal -> Signal -> Signal` transform, but it returns `SignalProducer -> SignalProducer` and thus the return type is not matched. It doesn't seem there is an overload, and IMO the compiler doesn't complain because the argument type and return type both matches. I tried to reproduce it (http://pastebin.com/gLy1dwHt) in playground and it crashes.